pkg/config/remote/service/client_predicates.go

```diff
@@ -0,0 +1,135 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/Masterminds/semver"
+	"github.com/theupdateframework/go-tuf/data"
+)
+
+type tracerPredicates struct {
+	RuntimeID     *string `json:"runtime-id,omitempty"`
+	Service       *string `json:"service,omitempty"`
+	Environment   *string `json:"environment,omitempty"`
+	AppVersion    *string `json:"app-version,omitempty"`
+	TracerVersion *string `json:"tracer-version,omitempty"`
+	Language      *string `json:"language,omitempty"`
+}
+
+type clientPredicates struct {
+	Version    int                 `json:"version,omitempty"`
+	Predicates []*tracerPredicates `json:"predicates,omitempty"`
+}
+
+type DirectorTargetsCustomMetadata struct {
+	Predicates *clientPredicates `json:"tracer-predicates,omitempty"`
+}
+
+// Given the hostname and state will parse predicates and execute them
+// It will return a list ConfigPointers
+func executeClientPredicates(
+	client *pbgo.Client,
+	directorTargets data.TargetFiles,
+) ([]string, error) {
+	configs := make([]string, 0)
+
+	for path, meta := range directorTargets {
+		predicates, err := parsePredicates(meta.Custom)
+		if err != nil {
+			return nil, err
+		}
+
+		var matched bool
+		nullPredicates := predicates == nil || predicates.Predicates == nil
```

Comment:
Go style Nit: You don't typically see Go code that does the assignment as part of the boolean expression.

```suggestion
        nullPredicates := predicates == nil
		nullPredicates || predicates.Predicates == nil
```

Reply:
I am going without this one, the variable should tell me if predicates is not null and if it also contains actual predicates. 
I changed the name of predicates to clientPredicates